tests/test_teststats.py

```diff
@@ -0,0 +1,130 @@
+
+    with caplog.at_level(logging.WARNING, 'pyhf.infer.test_statistics'):
+        pyhf.infer.test_statistics.qmu(mu, data, pdf, init_pars, par_bounds)
+        assert 'WARNING  qmu test statistic used for fit' in caplog.text
+        caplog.clear()
+
+
+def test_qmu_tilde(caplog):
+    mu = 1.0
+    pdf = pyhf.simplemodels.hepdata_like([6], [9], [3])
+    data = [9] + pdf.config.auxdata
+    init_pars = pdf.config.suggested_init()
+    par_bounds = pdf.config.suggested_bounds()
+
+    par_bounds[pdf.config.poi_index] = [-10, 10]
+    with caplog.at_level(logging.WARNING, 'pyhf.infer.test_statistics'):
+        pyhf.infer.test_statistics.qmu_tilde(mu, data, pdf, init_pars, par_bounds)
+        assert 'WARNING  qmu_tilde test statistic used for fit' in caplog.text
+        caplog.clear()
+
+
+def test_tmu(caplog):
+    mu = 1.0
+    pdf = pyhf.simplemodels.hepdata_like([6], [9], [3])
+    data = [9] + pdf.config.auxdata
+    init_pars = pdf.config.suggested_init()
+    par_bounds = pdf.config.suggested_bounds()
+    with caplog.at_level(logging.WARNING, 'pyhf.infer.test_statistics'):
+        pyhf.infer.test_statistics.tmu(mu, data, pdf, init_pars, par_bounds)
+        assert 'WARNING  tmu test statistic used for fit' in caplog.text
+        caplog.clear()
+
+
+def test_tmu_tilde(caplog):
+    mu = 1.0
+    pdf = pyhf.simplemodels.hepdata_like([6], [9], [3])
+    data = [9] + pdf.config.auxdata
+    init_pars = pdf.config.suggested_init()
+    par_bounds = pdf.config.suggested_bounds()
+
+    par_bounds[pdf.config.poi_index] = [-10, 10]
+    with caplog.at_level(logging.WARNING, 'pyhf.infer.test_statistics'):
+        pyhf.infer.test_statistics.tmu_tilde(mu, data, pdf, init_pars, par_bounds)
+        assert 'WARNING  tmu_tilde test statistic used for fit' in caplog.text
+        caplog.clear()
+
+
+def test_no_poi_test_stats(caplog):
+    # FIXME: Can reduce the model more
+    spec = {
+        'channels': [
+            {
+                'name': 'channel',
+                'samples': [
+                    {
+                        'name': 'goodsample',
+                        'data': [10.0],
+                        'modifiers': [
+                            {
+                                'type': 'normsys',
+                                'name': 'shape',
+                                'data': {"hi": 0.5, "lo": 1.5},
+                            }
+                        ],
+                    },
+                ],
+            }
+        ]
+    }
+    model = pyhf.Model(spec, poi_name=None)
+
+    test_poi = 1.0
+    data = [12] + model.config.auxdata
+    init_pars = model.config.suggested_init()
+    par_bounds = model.config.suggested_bounds()
+
+    with pytest.raises(pyhf.exceptions.UnspecifiedPOI):
+        with caplog.at_level(logging.DEBUG, 'pyhf.infer.test_statistics'):
+            pyhf.infer.test_statistics.qmu(test_poi, data, model, init_pars, par_bounds)
+            # FIXME: This isn't implimented correctly
```

Comment:
This FIXME needs to get taken care of

Reply:
@kratsg I'll look at this more tonight, but if you have an example of why the assert for `caplog.text` isn't working as expected (or isn't being run?) that could be helpful.

Reply:
Found the answer in the pytest docs: https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions